bin/liquidator/src/rebalance.rs

```diff
@@ -423,6 +409,301 @@ impl Rebalancer {
+            .await?;
+
+        let mut ixs = PreparedInstructions::new();
+        ixs.append(create_or_replace_ixs);
+        ixs.append(cancel_ixs);
+        ixs.append(place_order_ixs);
+        ixs.append(seq_check_ixs);
+
+        let txsig = self
+            .mango_client
+            .send_and_confirm_owner_tx(ixs.to_instructions())
+            .await?;
+
+        info!(
+            %txsig,
+            "placed order for {} {} at price = {}",
+            token.native_to_ui(I80F48::from(native_amount)),
+            token.name,
+            limit_price,
+        );
+        if !self.refresh_mango_account_after_tx(txsig).await? {
+            return Ok(());
+        }
+
+        Ok(())
+    }
+
+    async fn settle_and_close_all_openbook_orders(&self) -> anyhow::Result<()> {
+        let account = self.mango_account()?;
+
+        for x in Self::shuffle(account.active_serum3_orders()) {
+            let token = self.mango_client.context.token(x.base_token_index);
+            let quote = self.mango_client.context.token(x.quote_token_index);
+            let market_index = x.market_index;
+            let market = self
+                .mango_client
+                .context
+                .serum3_markets
+                .get(&market_index)
+                .expect("no openbook market found");
+            self.settle_and_close_openbook_orders(&account, token, &market_index, market, quote)
+                .await?;
+        }
+        Ok(())
+    }
+
+    async fn settle_and_close_openbook_orders(
+        &self,
+        account: &Box<MangoAccountValue>,
+        token: &TokenContext,
+        market_index: &Serum3MarketIndex,
+        market: &Serum3MarketContext,
+        quote: &TokenContext,
+    ) -> anyhow::Result<()> {
+        let open_orders_opt = account
+            .serum3_orders(*market_index)
+            .map(|x| x.open_orders)
+            .ok();
+
+        if open_orders_opt.is_none() {
+            return Ok(());
+        }
+
+        let open_orders = open_orders_opt.unwrap();
+        let oo_acc = self.account_fetcher.fetch_raw(&open_orders)?;
+        let oo = serum3_cpi::load_open_orders_bytes(oo_acc.data())?;
+        let oo_slim = OpenOrdersSlim::from_oo(oo);
+
+        if oo_slim.native_base_reserved() != 0 || oo_slim.native_quote_reserved() != 0 {
+            return Ok(());
+        }
+
+        let settle_ixs = PreparedInstructions::from_single(
+            self.mango_client
+                .serum3_settle_funds_instruction(market, token, quote, open_orders),
+            self.mango_client
+                .context
+                .compute_estimates
+                .cu_per_serum3_order_cancel, // TODO FAS
+        );
+
+        let close_ixs = self
+            .mango_client
+            .serum3_close_open_orders_instruction(*market_index);
+
+        let mut ixs = PreparedInstructions::new();
+        ixs.append(settle_ixs);
+        ixs.append(close_ixs);
```

Comment:
the ordering is odd, you should close open orders (making reserved funds free) and then settle (transfering the free funds into the account)